dao/accumulo.rya/src/main/java/org/apache/rya/accumulo/AbstractAccumuloRdfConfigurationBuilder.java

```diff
@@ -0,0 +1,209 @@
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.rya.accumulo;
+
+import org.apache.rya.api.RdfCloudTripleStoreConfiguration;
+import org.apache.rya.api.RdfCloudTripleStoreConfigurationBuilder;
+
+import com.google.common.base.Preconditions;
+
+/**
+ * This builder class will set all of the core Accumulo-backed Rya configuration
+ * parameters. Any builder extending this class will have setter methods for all
+ * of the necessary parameters to connect to an Accumulo backed Rya instance.
+ *
+ * @param <B>
+ *            - builder returned by setter methods extending this class
+ * @param <C>
+ *            - configuration object constructed by the builder extending this
+ *            class
+ */
+public abstract class AbstractAccumuloRdfConfigurationBuilder<B extends AbstractAccumuloRdfConfigurationBuilder<B, C>, C extends AccumuloRdfConfiguration>
+        extends RdfCloudTripleStoreConfigurationBuilder<B, C> {
+
+    private String user;
+    private String pass;
+    private String zoo = "";
+    private String instance;
+    private boolean useMock = false;
+    private boolean usePrefixHashing = false;
+    private boolean useComposite = false;
+    private boolean useSelectivity = false;
+
+    protected static final String ACCUMULO_USER = "accumulo.user";
+    protected static final String ACCUMULO_PASSWORD = "accumulo.password";
+    protected static final String ACCUMULO_INSTANCE = "accumulo.instance";
+    protected static final String ACCUMULO_AUTHS = "accumulo.auths";
+    protected static final String ACCUMULO_VISIBILITIES = "accumulo.visibilities";
+    protected static final String ACCUMULO_ZOOKEEPERS = "accumulo.zookeepers";
+    protected static final String ACCUMULO_RYA_PREFIX = "accumulo.rya.prefix";
+    protected static final String USE_INFERENCE = "use.inference";
+    protected static final String USE_DISPLAY_QUERY_PLAN = "use.display.plan";
+    protected static final String USE_MOCK_ACCUMULO = "use.mock";
+    protected static final String USE_PREFIX_HASHING = "use.prefix.hashing";
+    protected static final String USE_COUNT_STATS = "use.count.stats";
+    protected static final String USE_JOIN_SELECTIVITY = "use.join.selectivity";
+
+    /**
+     * Sets Accumulo user. This is a required parameter to connect to an
+     * Accumulo Rya Instance
+     * 
+     * @param user
+     *            - Accumulo user
+     */
+    public B setAccumuloUser(String user) {
+        Preconditions.checkNotNull(user, "User cannot be null.");
```

Comment:
These checks should happen within the build function when delegated to the object they are building, not a builder's set function. Don't duplicate the logic.